src/rSW_Model.c

```diff
@@ -168,3 +169,13 @@ void onSet_SW_MDL(SEXP SW_MDL) {
 
 	UNPROTECT(5);
 }
+
+/**
+ * @brief Interface between R and C to clear the simulation model
+ *
+ * @note Called to prevent memory leaks if SOILWAT2 fails when called
+ *       from a R-side function
+*/
+void rSW2_clearModel(void) {
+	SW_CTL_clear_model(swTRUE, &SoilWatAll, &PathInfo);

```

Comment:
Why does `rSW2_clearModel()` call `SW_CTL_clear_model()` with `full_reset = swTRUE` - even though the `p_OUT` allocation/de-allocation is handled by rSOILWAT2 and not SOILWAT2? See the call to `SW_CTL_clear_model()` in `start()`.